app/src/main/java/com/example/plantsapp/presentation/ui/worker/ReschedulingWorker.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.plantsapp.presentation.ui.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.example.plantsapp.domain.repository.PlantsRepository
+import com.example.plantsapp.domain.repository.TasksRepository
+import com.example.plantsapp.domain.usecase.GetTasksForPlantAndDateUseCase
+import com.example.plantsapp.presentation.ui.utils.minusDays
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import java.util.Date
+
+@HiltWorker
+class ReschedulingWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val plantsRepository: PlantsRepository,
+    private val getTasksForPlantAndDateUseCase: GetTasksForPlantAndDateUseCase,
+    private val tasksRepository: TasksRepository
+) : CoroutineWorker(context, workerParams) {
+
+    override suspend fun doWork(): Result {
+
+        plantsRepository.fetchPlants()
```

Comment:
![image](https://user-images.githubusercontent.com/23121846/144112268-ff9506dc-d3d2-4d47-acec-ecc8dee27132.png)
Im not 100% sure, but, maybe this will be more readable